[BUG] FILENAME_FORMAT_REMOVE_NONE with empty first path

### Description

I have `PAPERLESS_FILENAME_FORMAT` set to `{{ owner_username }}/{{ created_year }}/{{ correspondent }}/{{ title }}`. I also have `PAPERLESS_FILENAME_FORMAT_REMOVE_NONE` enabled.

With 2.13.0 and the new template handling, whenever `owner_username` is not set (which is the default in our instance), it gets removed from the path -- with the slash intact. That is, the path get rendered as `/{{ created_year }}/...` and paperless is thus trying to write to root, which is disallowed.

Looking at [file_handling.py](https://github.com/paperless-ngx/paperless-ngx/blob/b2e9f3195a38d387f7fabd24744d092e7800964e/src/documents/file_handling.py#L110) this is no surprise, as the slashes are only removed when they occur on both sides. Probably, something like `rendered_filename.removeprefix("-none-/")` needs to be added.

### Steps to reproduce

1. Have `PAPERLESS_FILENAME_FORMAT_REMOVE_NONE` enabled
2. Have a filename format set to a template where the first path resolves to `""`/`"-none"`
3. Add a document and see it fail, because it tries to write to root

### Webserver logs

```bash
[2024-10-27 14:43:58,087] [ERROR] [paperless.consumer] The following error occurred while storing document ADS_1100W_006904.pdf after parsing: [Errno 13] Permission denied: '/2024'

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 327, in main_wrap
    raise exc_info[1]
  File "/usr/src/paperless/src/documents/consumer.py", line 565, in run
    create_source_path_directory(document.source_path)
  File "/usr/src/paperless/src/documents/file_handling.py", line 11, in create_source_path_directory
    os.makedirs(os.path.dirname(source_path), exist_ok=True)
  File "<frozen os>", line 225, in makedirs
PermissionError: [Errno 13] Permission denied: '/2024'
```


### Browser logs

_No response_

### Paperless-ngx version

2.13.0

### Host OS

Ubuntu 24.04.1 (aarch64)

### Installation method

Docker - official image

### System status

_No response_

### Browser

_No response_

### Configuration changes

_No response_

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

--- a/src/documents/file_handling.py
+++ b/src/documents/file_handling.py
@@ -111,6 +111,7 @@ def generate_filename(
             rendered_filename = rendered_filename.replace("/-none-/", "/")
             rendered_filename = rendered_filename.replace(" -none-", "")
             rendered_filename = rendered_filename.replace("-none-", "")
+            rendered_filename = rendered_filename.strip(os.sep)
 
         rendered_filename = rendered_filename.replace(
             "-none-",

--- a/src/documents/tests/test_file_handling.py
+++ b/src/documents/tests/test_file_handling.py
@@ -960,17 +960,27 @@ class TestFilenameGeneration(DirectoriesMixin, TestCase):
             - the generated filename uses the defined storage path for the document
             - the generated filename does not include "none" in the place undefined field
         """
+        sp = StoragePath.objects.create(
+            path="TestFolder/{{asn}}/{{created}}",
+        )
         doc = Document.objects.create(
             title="does not matter",
             created=timezone.make_aware(datetime.datetime(2020, 6, 25, 7, 36, 51, 153)),
             mime_type="application/pdf",
             pk=2,
             checksum="2",
-            storage_path=StoragePath.objects.create(
-                path="TestFolder/{{asn}}/{{created}}",
-            ),
+            storage_path=sp,
         )
         self.assertEqual(generate_filename(doc), "TestFolder/2020-06-25.pdf")
+
+        # Special case, undefined variable, then defined at the start of the template
+        # This could lead to an absolute path after we remove the leading -none-, but leave the leading /
+        # -none-/2020/ -> /2020/
+        sp.path = (
+            "{{ owner_username }}/{{ created_year }}/{{ correspondent }}/{{ title }}"
+        )
+        sp.save()
+        self.assertEqual(generate_filename(doc), "2020/does not matter.pdf")
 
     def test_multiple_doc_paths(self):
         """

